options.go

```diff
@@ -83,6 +83,7 @@ type TransportOptions struct {
 	ShardKey         string            `long:"sk" description:"The shard key is a transport header that clues where to send a request within a clustered traffic group."`
 	Jaeger           bool              `long:"jaeger" description:"Use the Jaeger tracing client to send Uber style traces and baggage headers"`
 	TransportHeaders map[string]string `short:"T" long:"topt" description:"Transport options for TChannel, protocol headers for HTTP"`
+	HTTPMethod       string            `long:"http-method" description:"The HTTP method to use"`
```

Comment:
Should we mention that it defaults to POST here? Also, that this only applies if the transport is HTTP?

Reply:
go-flags has this useful feature where if you set the defaults before you parse the struct, it shows them automatically, e.g., from the help output:
```
      --http-method=             The HTTP method to use (default: POST)
```

It's a little nicer to use that rather than `default:"POST"` since the default won't be applied if you create the command without parsing using go-flags. With the `newCommand` approach, the main package tests can just set Go fields and avoid parsing, which simplifies tests and provides compiler verification that the field name is correct.

Reply:
Ah, I didn't realize go-flags will show the defaults you've set below.